src/types/dtype.rs

```diff
@@ -281,8 +281,12 @@ pub(crate) fn get_dtype_for_column<DT: CellType + Debug + DataType>(
     column_types.remove(&DType::Null);
 
     if column_types.is_empty() {
-        // If no type apart from NULL was found, it's a NULL column
-        Ok(DType::Null)
+        // If no type apart from NULL was found, fallback to string except if the column is empty
+        if start_row == end_row {
+            Ok(DType::Null)
+        } else {
+            Ok(DType::String)
+        }
```

Comment:
Logic LGTM, but I'd like to have a log message in that case, to avoid bug reports regarding this behaviour. Seems like  a good opportunity to introduce https://pyo3.rs/v0.22.3/ecosystem/logging WDYT ?